include/deal.II/numerics/data_out.h

```diff
@@ -274,14 +274,14 @@ class DataOut : public DataOut_DoFData<DH, DH::dimension, DH::space_dimension>
    * pair that is locally owned. If this object operates on a non-distributed
    * triangulation, the result equals what first_cell() returns.
    */
-  cell_iterator first_locally_owned_cell ();
+  virtual cell_iterator first_locally_owned_cell ();
```

Comment:
Ah, I see what you mean. I think in that case you probably simply want to change the implementation to make sure you only call `is_locally_owned` if the cell is active, and the corresponding condition if the cell is not active. But I think that belongs in the implementation of the existing function.